client/my-sites/upgrades/verify-email-forward/controller.jsx

```diff
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { Base64 } from 'js-base64';
+
+/**
+ * Internal dependencies
+ */
+import i18nUtils from 'lib/i18n-utils';
+import VerifyEmailForward from './verify-email-forward';
+
+export function verifyEmailForward( context, next ) {
+	const { mailbox, destination } = JSON.parse( Base64.decode( context.params.mailbox_base64 ) );
```

Comment:
Some error checking here would be good - we've had email clients mangle our URLs (and/or users when copy and pasting), so I can imagine that `mailbox_base64` won't be decoded successfully in all cases (or it won't decode to a proper JSON). We should show a helpful notice/view in such a case - maybe hinting them to try to verify if the URL has been completely copied and pasted, etc.
It's not a blocker right now, of course, something to think for the next iterations.